api/register/swagger-api_register.yaml

```diff
@@ -0,0 +1,72 @@
+openapi: 3.0.1
+info:
+  title: Swagger Cloudtrust Register
+  description: 'Self-register API for Cloudtrust.'
+  version: 1.0.0
+servers:
+- url: http://localhost:8855
+tags:
+- name: Register
+  description: Self registering of a user
+paths:
+  /register/realm/{realm}/user:
+    post:
+      tags:
+      - Register
+      summary: Create a user
+      security:
+        - BasicAuth: [recaptcha]
+      parameters:
+      - name: realm
+        in: path
+        description: realm name (not id!)
+        required: true
+        schema:
+          type: string
+      requestBody:
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/User'
+      responses:
+        200:
+          description: successful operation. Returns the generated username
+          content:
+            application/json:
+              schema:
+                type: string
+components:
+  schemas:
+    User:
+      type: object
+      properties:
+        gender:
+          type: string
```

Comment:
You could add which of these properties are required 
required:
  - firstname
  - lastname

Reply:
Done... but I'm not sure required properties always have sens. I think it depends on the context (if you want to update a User, you don't need to have all properties set)